rwengine/src/script/ScriptFunctions.hpp

```diff
@@ -43,13 +71,59 @@ inline bool objectInBounds(GameObject* object,
 	        p.y <= max.y && p.z <= max.z);
 }
 template <class Tobj, class Tvec>
-bool objectInArea(const ScriptArguments& args, Tobj* object, const Tvec& min, const Tvec& max, bool marker)
+bool objectInArea(const ScriptArguments& args, const Tobj& object, const Tvec& min, const Tvec& max, bool marker, bool preconditions = true)
+{
+	if (marker) {
+		auto center = (min+max)/2.f;
+		auto radius = glm::abs(max-min);
+		drawAreaIndicator(args, center, radius);
+	}
+	return preconditions && objectInBounds(object, glm::min(min, max), glm::max(min, max));
+}
+
+inline bool objectInSphere(GameObject* object,
+                           const ScriptVec2& center,
+                           const ScriptVec2& radius)
+{
+	auto p = glm::vec2(object->getPosition());
```

Comment:
Do we have other ways to swizzle this? I think there are more obvious ways to extract `.xy()` ?